chain/mempool.go

```diff
@@ -408,48 +390,156 @@ func (m *mempool) UpdateMempoolTxes() []*wire.MsgTx {
 			continue
 		}
 
-		eg.Go(func() error {
-			// Grab full mempool transaction from hash.
-			tx := m.getRawTxIgnoreErr(txHash)
+		newTxids = append(newTxids, txHash)
+	}
+
+	// Now, we clear our internal mempool of any unmarked transactions.
+	// These are all the transactions that we still have in the mempool but
+	// that were not returned in the latest GetRawMempool query.
+	m.DeleteUnmarked()
+
+	// Fetch the raw transactions in batch.
+	txesToNotify, err := m.batchGetRawTxes(newTxids, true)
+	if err != nil {
+		log.Error("Batch getrawtransaction got %v", err)
+	}
+
+	return txesToNotify
+}
+
+// getRawMempool returns all the raw transactions found in mempool.
+func (m *mempool) getRawMempool() ([]*chainhash.Hash, error) {
+	// Create an async request and send it immediately.
+	result := m.client.GetRawMempoolAsync()
```

Comment:
What overhead are you referring to? I think under the hood `GetRawMempool` behaves the same as it calls `GetRawMempoolAsync` in `btcd`. Also I tried to use `GetRawMempool` but it doesn't work for batch client.

Reply:
Just a suspicion, but since I don't really know how the async RPC works under the hood I can't go into more detail. And if there is no `GetRawMempool` call on batch clients, then I guess that settles it anyway.